src/main/java/org/elasticsearch/index/query/LimitQueryBuilder.java

```diff
@@ -31,12 +31,19 @@
 public class LimitQueryBuilder extends QueryBuilder {
 
     public static final String NAME = "limit";
-    private final int limit;
+    private final Integer limit;
```

Comment:
not loving this change, maybe just use -1 or some other value rather than null and leave int?